addons/mail/static/src/js/wysiwyg.js

```diff
@@ -0,0 +1,97 @@
+
+import { renderToElement } from "@web/core/utils/render";
+import { Wysiwyg } from "@web_editor/js/wysiwyg/wysiwyg";
+import { closestBlock, setCursorEnd } from "@web_editor/js/editor/odoo-editor/src/OdooEditor";
+import { patch } from "@web/core/utils/patch";
+import { MentionList } from "@mail/core/web/mention_list";
+import { url } from "@web/core/utils/urls";
+import { usePopover } from "@web/core/popover/popover_hook";
+
+patch(Wysiwyg.prototype, {
+    setup() {
+        super.setup();
+        if (this.props.options.recordInfo?.res_model === "mail.compose.message") {
+            this.mentionList = usePopover(MentionList, {
+                position: `bottom-start`,
+                onClose: () => {
+                    this.focus();
+                },
+            });
+        }
+    },
+    async startEdition() {
+        const res = await super.startEdition(...arguments);
+        // Only enable mention list in full chatter mode web editor
+        if (this.options.recordInfo?.res_model === "mail.compose.message") {
+            this.odooEditor.document.addEventListener(
+                "keydown",
+                this.triggerMentionList.bind(this),
+                true
+            );
+        }
+        return res;
+    },
+    destroy() {
+        super.destroy();
+        if (this.options.recordInfo?.res_model === "mail.compose.message" && this.odooEditor) {
+            this.odooEditor.document.removeEventListener(
+                "keydown",
+                this.triggerMentionList.bind(this),
+                true
+            );
+        }
+    },
+    triggerMentionList(ev) {
+        const selection = document.getSelection();
+        if (
+            this.isSelectionInEditable() &&
+            selection.isCollapsed &&
+            selection.rangeCount &&
+            !this.mentionList.isOpen &&
+            (ev.key === "@" || ev.key === "#")
+        ) {
+            this.stepBeforeMention = this.odooEditor._historySteps.length - 1;
+            const closest = closestBlock(this.odooEditor.document.getSelection().anchorNode);
+            this.mentionList.open(closest, {
+                type: ev.key === "@" ? "partner" : "channel",
+                onSelect: this.selectMention.bind(this),
+            });
+        }
+    },
+    selectMention: function (ev, option) {
+        this.mentionList.close();
+        const attrs = {};
+        if (option.partner) {
+            Object.assign(attrs, {
+                placeholder: "@" + option.label,
+                model: "res.partner",
+                class: "o_mail_redirect",
+                id: option.partner.id,
+            });
+        } else {
+            Object.assign(attrs, {
+                placeholder: "#" + option.label,
+                model: "discuss.channel",
+                class: "o_channel_redirect",
+                id: option.channel.id,
+            });
+        }
+
+        const mentionBlock = renderToElement("mail.wysiwyg_mention_link", {
```

Comment:
Nit-picking: would name template `mail.Wysiwyg.mentionLink` (follows template naming convention of `moduleName.ComponentName.partName`)